code/mission/import/xwinglib.cpp

```diff
@@ -582,5 +596,186 @@ bool XWingMission::load(XWingMission *m, const char *data)
+				break;
+			case 32:
+				noj->objectType = XWMObjectType::oj_Asteroid7;
+				break;
+			case 33:
+				noj->objectType = XWMObjectType::oj_Asteroid8;
+				break;
+			case 34:
+				noj->objectType = XWMObjectType::oj_Rock_World;
+				break;
+			case 35:
+				noj->objectType = XWMObjectType::oj_Gray_Ring_World;
+				break;
+			case 36:
+				noj->objectType = XWMObjectType::oj_Gray_World;
+				break;
+			case 37:
+				noj->objectType = XWMObjectType::oj_Brown_World;
+				break;
+			case 38:
+				noj->objectType = XWMObjectType::oj_Gray_World2;
+				break;
+			case 39:
+				noj->objectType = XWMObjectType::oj_Planet_and_Moon;
+				break;
+			case 40:
+				noj->objectType = XWMObjectType::oj_Gray_Crescent;
+				break;
+			case 41:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent1;
+				break;
+			case 42:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent2;
+				break;
+			case 43:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent3;
+				break;
+			case 44:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent4;
+				break;
+			case 45:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent5;
+				break;
+			case 46:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent6;
+				break;
+			case 47:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent7;
+				break;
+			case 48:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent8;
+				break;
+			case 49:
+				noj->objectType = XWMObjectType::oj_Death_Star;
+				break;
+			case 58:
+				noj->objectType = XWMObjectType::oj_Training_Platform1;
+				break;
+			case 59:
+				noj->objectType = XWMObjectType::oj_Training_Platform2;
+				break;
+			case 60:
+				noj->objectType = XWMObjectType::oj_Training_Platform3;
+				break;
+			case 61:
+				noj->objectType = XWMObjectType::oj_Training_Platform4;
+				break;
+			case 62:
+				noj->objectType = XWMObjectType::oj_Training_Platform5;
+				break;
+			case 63:
+				noj->objectType = XWMObjectType::oj_Training_Platform6;
+				break;
+			case 64:
+				noj->objectType = XWMObjectType::oj_Training_Platform7;
+				break;
+			case 65:
+				noj->objectType = XWMObjectType::oj_Training_Platform8;
+				break;
+			case 66:
+				noj->objectType = XWMObjectType::oj_Training_Platform9;
+				break;
+			case 67:
+				noj->objectType = XWMObjectType::oj_Training_Platform10;
+				break;
+			case 68:
+				noj->objectType = XWMObjectType::oj_Training_Platform11;
+				break;
+			case 69:
+				noj->objectType = XWMObjectType::oj_Training_Platform12;
+				break;
+			default:
+				return false;
+		}
+
+		switch (oj->object_formation) {
+			case 0:
+				noj->formation = XWMObjectFormation::ojf_FloorXY;
+				break;
+			case 1:
+				noj->formation = XWMObjectFormation::ojf_SideYZ;
+				break;
+			case 2:
+				noj->formation = XWMObjectFormation::ojf_FrontXZ;
+				break;
+			case 3:
+				noj->formation = XWMObjectFormation::ojf_Scattered;
+				break;
+			default:
+				return false;
+		}
+
+		switch (oj->object_goal) {
```

Comment:
There is no `object_goal` field in the byte structure (see the comment about `xwi_objectgroup`).  Instead `noj->objectGoal` must be set using the bit checks.